pkg/agent/controller/bgp/controller.go

```diff
@@ -931,3 +931,19 @@ func (c *Controller) updateBGPPeerPasswords(secret *corev1.Secret) {
 		}
 	}
 }
+
+// GetBGPPolicyInfo returns Name, RouterID, LocalASN and ListenPort of effective BGP Policy applied on the Node.
+func (c *Controller) GetBGPPolicyInfo() (string, string, int32, int32) {
+	var name, routerID string
+	var localASN, listenPort int32
+	bgpPolicy := c.getEffectiveBGPPolicy()
```

Comment:
IMO, calling this function here is incorrect. This function is meant to be called exclusively by `syncBGPPolicy`. It is possible that the `bgpPolicy` you will get here does not match the stored `c.bgpPolicyState`. My recommendation is to just add the policy name to `bgpPolicyState`, this way `c.bgpPolicyState` is all you need in this function.